### DISCUSSION TITLE
Anisotropic Diffusion for Cylindrical or Spherical Geometries

### DISCUSSION THREAD

**Discussion by AndrewFalkowski** [idaholab/moose#23044] [Q&A General] [answered]
How would one define anisotropic diffusivities that conform to a cylindrical or spherical geometry (see picture)? My initial thought is to create a material object that rotates the coefficient matrix based on element orientation or subdomain boundary orientation; however, I am not sure how to extract such information. I would appreciate some guidance on how to go about this.
![image](https://user-images.githubusercontent.com/3618750/210025362-b41ac413-2303-46c5-a5a0-5449879407d4.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think you ll want to have the "origin" and the type of the geometry (spherical/cylindrical) as parameters to an anisotropic diffusion coefficient.
From there you ll use the q_point attribute (taken at the local quadrature point index: `_q_point[_qp]`) to compute the local orientation.

Then when you re computing the qp value of the vector diffusion coefficient, you ll take into account the rotation.

You could have a look at the phase field module and see how they deal with grain orientation / euler angles for an example.

Guillaume
